airflow/utils/helpers.py

```diff
@@ -45,6 +45,15 @@
 S = TypeVar("S")
 
 
+def get_value_with_cache(cache: dict, key: str, insert_fn: Callable, *arg, **kwargs) -> Any:
+    """Returns value from cache or function"""
+    return_value = cache.get(key)
+    if not return_value:
+        return_value = cache[key] = insert_fn(*arg, **kwargs)
+
+    return return_value
```

Comment:
By using a lambda, we don't need to handle paramters here. (suggesting a rename of the function passed as well ? WDYT ?)

Also, `if not return_value` will trigger if the saved value is "falsy", which is the case for None, but also for empty lists/dict for instance, which can lead to surprising behavior. I think explicitly checking against None would be better here.

```suggestion
def get_value_with_cache(cache: dict, key: str, do_get_value: Callable) -> Any:
    """Returns value from cache or function"""
    return_value = cache.get(key)
    if return_value is None:
        return_value = cache[key] = do_get_value()

    return return_value
```

Reply:
I'll let you fix the test if you apply this change ;)

Reply:
Thanks for the feedback. I like the new name for the function! 
I'm using `if not return_value` because if the cache returns empty value, I think the best way is to call do_get_value to make sure that the right and latest value is used and not the cached value.

Reply:
ok makes sense then. That specific case doesn't really matter for the current application anyway.